### DISCUSSION TITLE
The manifest file name in Modular Authorization Models

### DISCUSSION THREAD

**Discussion by velmohan** [openfga/community#309] [Q&A] [answered]
I have noticed that if I do not name the manifest file as fga (like in fga.mod), VS Code validation does not run for any modules. Is this expected?

For example, in this sample store, if you change fga.mod to XYZ.mod and do some breaking change like changing the type name in core to user2, the errors are not shown by VS Code unless you change the manifest file back to fga.mod.

https://github.com/openfga/sample-stores/tree/main/stores/modular

**Comment by ewanharris** [comment, marked as answer]
Hey @velmohan,

Yes, we require that the manifest file to be named exactly `fga.mod`, similar to how Go requires the `go.mod` name or JS/npm requires `package.json`. By having a known name it means that we can simplify the validation and lookup of this file/modular models in general.

Do you have a usecase that requires you to rename this file?

**Reply by velmohan** [reply]
Thanks @ewanharris . That makes sense. Our usecase is that we have multiple authz model files in the same source repository folder. The model files are named after their project/team. We have had this structure since well before modular feature was introduced and we used to "merge" base modules using scripts. 

The new modular models feature means we do not need to use the script anymore and the structure is so flexible and intuitive. I will find a workaround for the name requirement of the `mod` files for validation.
